src/test/java/uk/gov/hmcts/reform/civil/utils/HearingUtilsTest.java

```diff
@@ -63,20 +67,55 @@ void shouldReturnFee7_whenGivenAnyClaimFee() {
     }
 
     @Test
-    void shouldReturnFormattedFee_whenGivenAnyClaimFee() {
+    void shouldReturnFormattedFee_whenGivenMidClaimFee() {
         assertThat(HearingUtils.formatHearingFee(
             Fee.builder().calculatedAmountInPence(new BigDecimal(34600)).build())).isEqualTo("£346");
     }
 
     @Test
-    void shouldReturnFormattedFee2_whenGivenAnyClaimFee() {
+    void shouldReturnFormattedFee_whenGivenBigClaimFee() {
         assertThat(HearingUtils.formatHearingFee(
             Fee.builder().calculatedAmountInPence(new BigDecimal(132000)).build())).isEqualTo("£1,320");
     }
 
     @Test
-    void shouldReturnFormattedFee3_whenGivenAnyClaimFee() {
+    void shouldReturnFormattedFee_whenGivenSmallClaimFee() {
         assertThat(HearingUtils.formatHearingFee(
             Fee.builder().calculatedAmountInPence(new BigDecimal(5000)).build())).isEqualTo("£50");
     }
+
+    @Test
+    void shouldReturnNull_when0ClaimFee() {
+        assertThat(HearingUtils.formatHearingFee(
+            Fee.builder().calculatedAmountInPence(new BigDecimal(0)).build())).isNull();
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {34600, 132000, 5000})
+    void shouldReturnFormattedFee_whenGivenAnyClaimFee(int amount) {
+        BigDecimal feeAmount = new BigDecimal(amount);
+        assertThat(HearingUtils.formatHearingFee(
+            Fee.builder().calculatedAmountInPence(feeAmount).build())).isNotEmpty();
+    }
+
+    @ParameterizedTest
+    @EnumSource(HearingDuration.class)
+    void shouldReturnHearingDuration_whenGivenAnyHearingDuration(HearingDuration hearingDuration) {
+        assertThat(HearingUtils.formatHearingDuration(hearingDuration)).isNotEmpty();
+    }
+
+    @Test
+    void shouldReturnNull_whenNotAllowedTime() {
+        assertThat(HearingUtils.getHearingTimeFormatted("50000")).isNull();
+    }
```

Comment:
Also strings with less than 4 characters or with text other than numbers are not allowed here.

Consider the following test instead:
```
    @ParameterizedTest
    @ValueSource(strings = {"000", "50000", "08:00", "8:00", "12:00", "23:00"})
    void shouldReturnNull_whenNotAllowedTime(String input) {
        assertThat(HearingUtils.getHearingTimeFormatted(input)).isNull();
    }
```

This test will capture also those other possible situations (even though they shouldn't happen) and will fail the test appropriately as we don't deal with them. A change to the `getHearingTimeFormatted` function goes along with this test to ensure those additional disallowed inputs are considered.